### ISSUE TITLE
Fix S2589 FP: || and property patterns

### ISSUE THREAD

**Issue by jnyrup** [SonarSource/sonar-dotnet#7096]
### Description

While rewriting some older code to using pattern matching it triggered S2589 when using `||` together with a property pattern.
```cs
obj is A { Value: true } || obj is B { Value: true };
```

When rewriting the disjunction with a logical pattern using ` or ` S2589 disappeared despite both lowered and compiled code are [identical](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0ATEBqAPgAQGYACASwDsAXGKAMwEMwZiBJYgb2IF8BYAKH5Fi+AEzEAgsRCsO3foJKjiAISkzOvAXyUBhfu37EjwksAgQANsQBi5gBRsIwAFYBKYgF4AfMSfOyAM4SslzEuLi+LoEqIQDcxAD0CcQAyiIArAAcAJzyfMYmxGaWKvRQDpFunj5+0ZLhldHKsXkFQsVWyvQAXhV+7t6NpEGSGr5QMRotfFxAA).
```cs
obj is A { Value: true } or B { Value: true };
```

### Repro steps

```cs
public interface I { }

public class A : I { }

public class B : I { }

class C
{
    public bool Foo(I obj) => obj is A { } || obj is B { }; // Triggers S2589 on "obj is B { }"

    public bool Bar(I obj) => obj is A || obj is B; // No warnings

    public bool Baz(I obj) => obj is A { } or B { }; // No warnings
}
```

### Expected behavior

S2589 is not triggered for `Foo`

### Actual behavior

S2589 is triggered for `Foo`.

### Known workarounds

Rewrite `Foo` to `Baz`.

### Related information

* C#/VB.NET Plugins version: SonarLint for Visual Studio 2022 6.15.0.67199
* Visual Studio version: 2022 17.6.0 Preview 3.0
* MSBuild / dotnet version: .NET 6
* SonarScanner for .NET version (if used)
* Operating System: Windows 10

**Comment by pavel-mikula-sonarsource** [comment]
Hi @jnyrup,

Thank you for reporting this case. I confirm it as FP. We plan to reimplement the rule soon and that should address the problem.

**Comment by mary-georgiou-sonarsource** [comment]
Fixed by https://github.com/SonarSource/sonar-dotnet/pull/7750
